include/deal.II/lac/affine_constraints.templates.h

```diff
@@ -334,11 +312,11 @@ AffineConstraints<number>::close()
     std::vector<size_type> new_lines(lines_cache.size(),
                                      numbers::invalid_size_type);
     size_type              counter = 0;
-    for (typename std::vector<ConstraintLine>::const_iterator line =
-           lines.begin();
-         line != lines.end();
-         ++line, ++counter)
-      new_lines[calculate_line_index(line->index)] = counter;
+    for (const ConstraintLine &line : lines)
+      {
+        new_lines[calculate_line_index(line.index)] = counter;
+        ++counter;
```

Comment:
You could shorten this to `new_lines[calculate_line_index(line.index)] = counter++;` for better performance. :wink:

Reply:
I have trouble remembering the semantics of ` a = x++` vs `a = ++x` so yes, I would also prefer to keep them separate :)

Reply:
Sure, that's fine with me.